### ISSUE TITLE
Laura's review and questions

### ISSUE THREAD

**Issue by lauracowen** [OpenLiberty/guide-jms-intro#72]
I've drafted some suggested text for the abstract and intro sections:

>Learn how to produce and consume messages to communicate between Java microservices in a standard way by using the Jakarta Messaging API with the embedded Liberty Messaging Server or an external messaging server, IBM MQ. 

**What you’ll learn** 

 >You’ll learn how to communicate between Java web services when one service is producing a continuous stream of <link:microprofile-reactive-messaging asynchronous messages> (or events) to be consumed by other services, rather than just sending and receiving individual requests for data. You will also learn how to use a messaging server and client to manage the production and consumption of the messages by the services. 

>In this guide, you will first use the embedded Liberty Messaging Server to produce messages, then you will (optionally) switch to using an external messaging server to produce the messages, in this case, <IBM MQ>. You might use an external messaging server if it is important to your application (e.g. during a bank transfer in a banking application) that none of your messages are lost if there is a system outage. 

 >You will learn how to write your Java application using the Jakarta Messaging API which provides a standard way to produce and consume messages in Java applications, regardless of which messaging server your application will ultimately use. 

>The application in this guide consists of two microservices, system and inventory. Every 15 seconds, the system microservice computes and publishes a message that contains system’s current CPU and memory load usage. The inventory microservice subscribes to that information at the `/systems` REST endpoint so that it can keep an updated list of all the systems and their current system loads. This application is different from the application in the <rest-intro guide> in which the Inventory service requests the system properties from the System service only when needed, and the System service responds with the data only when it receives a request for it. 

>You’ll create the `system` and `inventory` microservices using the Jakarta Messaging API to produce and consume the messages using the embedded Liberty Messaging Server. 

[diagram] 

>You will then, optionally, reconfigure the application, without otherwise changing the Java code, to use an external IBM MQ messaging server instead. 

**Additional prerequisites** 

* Remove the first para; it’s really hard to parse, then:

   >To complete the optional **Using IBM MQ** section in this guide, install Docker if it not already installed. You will use an IBM MQ container as the external messaging server. For installation instructions...[etc] 

**Try what you’ll build** 

After the first command add some explanation:

  >To try out the application, first go to the `finish` directory and run the following Maven goal to build and install the `models` module. The `models` module contains the data class `SystemLoad` for both the `system` and `inventory` services to use.  

**Creating the consumer in the inventory microservice** 

* “for monitoring the jms/InventoryQueue queue” Rephrase as “to monitor the `jms/InventoryQueue` queue.” 
* “Override the onMessage() method that processes the incoming messages, updates the inventory by using the InventoryManager bean, and logs the action. Use the SystemLoad.fromJson() method to convert the JSON message string to the SystemLoad object.”  I’m not sure what this sentence means. I think it’s quite long to parse but I’m not sure it says why it’s overriding the behaviour. Is the list of “updates the inventory...updates the method...logs the action” meant to describe what the override behaviour is? I don’t think the sentence reads quite like that at the moment; it currently implies that that that’s the non-overridden behaviour. 

 * “Enable the inventory service with an embedded messaging server to monitor the message queue.”  I’m not sure what this sentence is saying. Should it say “Next, configure the inventory service with an embedded messaging server to monitor the message queue.”?  I also found it a bit odd that the messaging server would monitor the message queue; wouldn’t that be the messaging client? 

 * “Configure a jmsConnectionFactory element to use the InventoryCM” Should this say “Configure a jmsConnectionFactory element to use the InventoryCM connection factory”? 

 * “the Jakarta Messaging service”  Is Jakarta Messaging actually a service? Or is it an API that talks to any messaging engine? 

* “running in the Jakarta Messaging service by using the send() API”  Should that say “...by using the send() method” (not “API”)? Is Jakarta Messaging really a service? Or just an interface to whatever messaging engine you’ve got. 

* “press enter/return key to restart the Liberty instance.”  Suggest appending a reason for this:
    > “press enter/return key to restart the Liberty instance so that Liberty reads the configuration changes.” 

 
**Using IBM MQ – Optional** 

* Suggest change this to
  > Using IBM MQ as the messaging server (optional)

* Then can you append a bit to the section intro to help orientate the reader before starting: 

  > ...to use IBM MQ container as the messaging server instead of the embedded Liberty Messaging Server.

* “If it builds successfully, you will see an image similar to” Suggest instead (for style):
  > When the container image is built, you see an image similar to...” 

* “If the IBM MQ container runs successfully” Similarly:
  > “When the IBM MQ container is running, you can access...” 

  Unless you’re going to also give instructions on troubleshooting why it hasn’t been successful, just assume it’s successful so that you don’t cast doubt on the instructions. 

* “when you deploy the application on production environment” Add “a”:
  > “when you deploy the application on a production environment” 

* “because they are no longer being required.” Delete “being”:
  > “because they are no longer required.” 

* “After you see the following message,” Style:
  > “When you see the following message,”

**Comment by gkwan-ibm** [comment]
close
- all comments were addressed
